eval.py

```diff
@@ -1,81 +1,119 @@
-""" Given model segmentation and radiologist ground truth, comapre the mIoU per task.
-    
-    Usage: python3 eval.py --phase valid --pred_path /path/to/segmentation json --save_dir /path/to/save/results
 """
+Given predicted and ground-truth segmentations, compute mIoU and confidence
+intervals for each of the 10 pathologies.
 
-from pycocotools import mask
-import pandas as pd
-from tqdm import tqdm
+Usage: python3 eval_miou.py --phase val --save_dir /path/to/save/results
+"""
+from argparse import ArgumentParser
 import json
+import numpy as np
+import pandas as pd
 from pathlib import Path
-from argparse import ArgumentParser
-from eval_helper import create_mask, iou_seg
+from PIL import Image
+from pycocotools import mask
+from tqdm import tqdm
+
 from eval_constants import LOCALIZATION_TASKS
 
 
-def compute_metrics(gt_dir, pred_dir):
+def calculate_iou(pred_mask, gt_mask, true_pos_only):
     """
-    Take in ground truth and prediction json (both encoded) and return iou for each image under each pathology
+    Calculate IoU score between two segmentation masks.
+
+    Args:
+        pred_mask (np.array): binary segmentation mask
+        gt_mask (np.array): binary segmentation mask
+    Returns:
+        iou_score (np.float64)
     """
-    with open(gt_dir) as f:
-        gt = json.load(f)
+    intersection = np.logical_and(pred_mask, gt_mask)
+    union = np.logical_or(pred_mask, gt_mask)
 
-    with open(pred_dir) as f:
-        pred = json.load(f)
+    # TODO: just want to make sure I understand this
+    if true_pos_only:
+        if np.sum(pred_mask) == 0 or np.sum(gt_mask) == 0:
+            iou_score = np.nan
+        else:
+            iou_score = np.sum(intersection) / (np.sum(union))
+    else:
+        if np.sum(gt_mask) == 0:
+            iou_score = np.nan
+        else:
+            iou_score = np.sum(intersection) / (np.sum(union))
 
-    ious = {}
+    return iou_score
+
+
+def get_ious(gt_path, pred_path, true_pos_only):
+    """
+    Returns IoU scores for each combination of CXR and pathology in gt_path and pred_path.
+
+    Args:
+        gt_path (str): path to ground-truth segmentation json file (encoded)
+        pred_path (str): path to predicted segmentation json file (encoded)
+        true_pos_only (bool): if true, TODO
```

Comment:
Can we add a bit more description here for what exactly is happening if this is true?